src/syscalls/native.rs

```diff
@@ -2,9 +2,21 @@ use crate::{ckb_constants::*, error::SysError};
 use core::ffi::CStr;
 
 #[cfg(target_arch = "riscv64")]
-#[link(name = "ckb-syscall")]
-extern "C" {
-    fn syscall(a0: u64, a1: u64, a2: u64, a3: u64, a4: u64, a5: u64, a6: u64, a7: u64) -> u64;
+#[naked_function::naked]
```

Comment:
It might be better to use a normal `asm!` block that takes input from `a0` to `a7` and outputs to `a0`, allowing the function to be inlined and avoid depending on the naked_function crate or nightly naked function feature.

Reply:
TBH I actually tried the exact step you mentioned, the code would be like following:

```
fn syscall(a0: u64, a1: u64, a2: u64, a3: u64, a4: u64, a5: u64, a6: u64, a7: u64) -> u64 {
  let mut ret = u64::MAX;
  asm!(
    "ecall",
    "mv {ret}, a0",
    ret = out(ret) _,
    in("a0") a0,
    in("a1") a1,
    in("a2") a2,
    in("a3") a3,
    in("a4") a4,
    in("a5") a5,
    in("a6") a6,
    in("a7") a7
  );
  ret
}
```

The problem with this path, is that each `ecall` will need a pairing `mv` instruction, since I couldn't find a way to tell Rust to use `a0` as the return value of the function. An extra variable is needed, together with an extra argument is needed here.

As a result, I use `naked_function` crate so as ro maintain the original behavior, which maintains non-inlineable `syscall` function in a separate assembly file. A better way would be just to duplicate `call` in each individual functions that calls into `syscall` but it might be better suited to another PR. Here we are maintaing the same behavior as before.

Reply:
Just use `inout("a0") a0`:

```rust
pub unsafe fn syscall(mut a0: u64, a1: u64, a2: u64, a3: u64, a4: u64, a5: u64, a6: u64, a7: u64) -> u64 {
    asm!( "ecall", inout("a0") a0, in("a1") a1, in("a2") a2, in("a3") a3, in("a4") a4, in("a5") a5, in("a6") a6, in("a7") a7 );
    a0
}
```

Reply:
Ahh I didn't know this behavior, thanks